### ISSUE TITLE
`master` build failing on Windows

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi#9014]
The last few `master` builds have failed on Windows with this:

```
        	Error:      	"{\n  \"plugins\": null,\n  \"host\": {\n    \"os\": \"Microsoft Windows Server 2022 Datacenter\",\n    \"version\": \"10.0.20348 Build 20348\",\n    \"arch\": \"x86_64\"\n  },\n  \"backend\": {\n    \"name\": \"fv-az448-481\",\n    \"url\": \"file://C:/Users/RUNNER~1/AppData/Local/Temp/test-env3127767626\",\n    \"user\": \"fv-az448-481\\\\runneradmin\"\n  },\n  \"currentStack\": null,\n  \"cliAbout\": {\n    \"version\": \"3.25.0-alpha.1645132829\",\n    \"goVersion\": \"go1.17.7\",\n    \"goCompiler\": \"gc\"\n  },\n  \"runtime\": {\n    \"language\": \"nodejs\",\n    \"executable\": \"C:\\\\hostedtoolcache\\\\windows\\\\node\\\\14.19.0\\\\x64\\\\node.exe\",\n    \"version\": \"v14.19.0\"\n  },\n  \"dependencies\": null,\n  \"errors\": [\n    \"Failed to get information about the plugin: failed to load language plugin nodejs: could not read plugin [D:\\\\a\\\\pulumi\\\\pulumi\\\\bin\\\\pulumi-language-nodejs.exe] stdout: EOF\",\n    \"Failed to get information about the Puluimi program's plugins: Could not find either C:\\\\Users\\\\RUNNER~1\\\\AppData\\\\Local\\\\Temp\\\\test-env3127767626\\\\yarn.lock or C:\\\\Users\\\\RUNNER~1\\\\AppData\\\\Local\\\\Temp\\\\test-env3127767626\\\\package-lock.json\",\n    \"Failed to get information about the current stack: No current stack\"\n  ]\n}\n" does not contain "go1.17.6"
```

E.g. https://github.com/pulumi/pulumi/actions/runs/1861124616.

**Comment by AaronFriel** [comment]
Error reports that the host OS is "Windows Server 2022 Datacenter", and that OS became the default recently. Our last passing test ran on Windows Server 2019:

https://github.com/pulumi/pulumi/runs/5224033778?check_suite_focus=true#step:1:3

**Comment by mikhailshilkov** [comment]
The latest run had a single error on Windows:

> CLI          \nVersion      3.25.0-alpha.1645217046\nGo Version   go1.17.7\nGo Compiler  gc\n\nHost     \nOS       Microsoft Windows Server 2022 Datacenter\nVersion  10.0.20348 Build 20348\nArch     x86_64\n\nThis project is written in nodejs (C:\\hostedtoolcache\\windows\\node\\14.19.0\\x64\\node.exe v14.19.0)\n\nBackend  \nName     fv-az448-481\nURL      file://C:/Users/RUNNER~1/AppData/Local/Temp/test-env1097509718\nUser     fv-az448-481\\runneradmin\n\nPulumi locates its logs in C:\\Users\\RUNNER~1\\AppData\\Local\\Temp by default\n" does not contain "go1.17.6

It looks like there was a mismatch in Go versions - 1.17.7 actual vs. 1.17.6 expected. I'll re-run it to be sure. Pulumi CLI is built with a different Go version than the one that runs our tests?

**Comment by AaronFriel** [comment]
@mikhailshilkov yep, that's it. I've created a draft PR here #9028 which does a few things to make our tests a bit more rigorous:

1. It moves our builds from macos-latest to ubuntu-latest, this was just something I thought was odd, to treat Mac as the primary build platform? Will revert this commit if it causes any regression.

2. It ensures our builders & runners check that the version of go in the runner cache is the latest that matches the version requested (currently 1.17.x).

3. It makes the test more resilient and more likely to pass on developer machines by making the test only check the major and minor version.

**Comment by AaronFriel** [comment]
Fixed with #9028.
